internal/controllers/extension_controller.go

```diff
@@ -183,8 +185,10 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 	if mediaType != catalogmetadata.MediaTypePlain {
 		// Set the TypeInstalled condition to Unknown to indicate that the resolution
 		// hasn't been attempted yet, due to the spec being invalid.
-		ext.Status.InstalledBundleResource = ""
-		setInstalledStatusConditionUnknown(&ext.Status.Conditions, fmt.Sprintf("bundle type %s not supported currently", mediaType), ext.GetGeneration())
+		if ext.Status.InstalledBundleResource == "" {
+			setInstalledStatusConditionUnknown(&ext.Status.Conditions, fmt.Sprintf("bundle type %s not supported currently", mediaType), ext.GetGeneration())
+		}
```

Comment:
I would only be expecting changes to the `InstalledBundleResource` and the `Installed` condition based on `Get`-ing the existing `App`.

Resolution and Installation are orthogonal. A failure computing a resolution shouldn't impact our ability to assess the installed state and vice versa.